tests/test_main.py

```diff
@@ -328,28 +339,76 @@ def test_packages_distributions_all_module_types(self):
         Test top-level modules detected on a package without 'top-level.txt'.
         """
         suffixes = importlib.machinery.all_suffixes()
+        filenames = list(
+            itertools.chain.from_iterable(
+                [
+                    f'top_level_{i}{suffix}',
+                    f'in_namespace_{i}/mod{suffix}',
+                    f'in_package_{i}/__init__.py',
+                    f'in_package_{i}/mod{suffix}',
+                ]
+                for i, suffix in enumerate(suffixes)
+            )
+        )
         fixtures.build_files(
             {
                 'all_distributions-1.0.0.dist-info': {
                     'METADATA': """
                         Name: all_distributions
                         Version: 1.0.0
                     """,
-                    'RECORD': ''.join(
-                        f'{i}-top-level{suffix},,\n'
-                        f'{i}-in-namespace/mod{suffix},,\n'
-                        f'{i}-in-package/__init__.py,,\n'
-                        f'{i}-in-package/mod{suffix},,\n'
-                        for i, suffix in enumerate(suffixes)
-                    ),
+                    'RECORD': ''.join(f'{fname},,\n' for fname in filenames),
                 },
             },
             prefix=self.site_dir,
         )
+        fixtures.build_files({fname: "" for fname in filenames}, prefix=self.site_dir)
```

Comment:
I've noticed these changes conflict with other changes to main. Also, they seem somewhat independent of the scope of this issue (though I understand they may need to adapt). My plan is this:

- Back out the two commits that affect this test, alleviating the conflicts.
- In main, refactor the behavior to (a) ensure files are created and (b) derive the 'RECORD' from the FilesDef.
- Probably at the same time, harmonize the behavior of file creation with that of `importlib_resources` by relying on `jaraco.path.build`.

I've started on the work to support these behaviors in https://github.com/jaraco/jaraco.path/commit/13732c82dbe0e86cd13fb9cd656bc4509a37c610.

Reply:
Done in #446.